repos/c2sm/packages/icon-exclaim/package.py

```diff
@@ -0,0 +1,92 @@
+from spack.pkg.c2sm.icon import Icon
+import shutil
+import os
+import re
+from collections import defaultdict
+import spack.error as error
+
+
+def validate_variant_dsl(pkg, name, value):
+    set_mutual_excl = set(['substitute', 'verify', 'serialize'])
+    set_input_var = set(value)
+    if len(set_mutual_excl.intersection(set_input_var)) > 1:
+        raise error.SpecError(
+            'Cannot have more than one of (substitute, verify, serialize) in the same build'
+        )
+
+
+class IconExclaim(Icon):
+    git = 'git@github.com:C2SM/icon-exclaim.git'
+
+    maintainers('jonasjucker', 'huppd')
+
+    version('develop', branch='icon-dsl', submodules=True)
+
+    # EXCLAIM-GT4Py specific features:
+    dsl_values = ('substitute', 'verify', 'serialize', 'nvtx')
+    variant('dsl',
+            default='none',
+            validator=validate_variant_dsl,
+            values=('none', ) + dsl_values,
+            description='Build with GT4Py dynamical core',
+            multi=True)
+
+    for x in dsl_values:
+        depends_on('icon4py', type="build", when=f"dsl={x}")
+        conflicts('^python@:3.9,3.11:', when='dsl={0}'.format(x))
+
+    def configure_args(self):
+        args = super().configure_args()
+
+        super_libs = args.pop()
+        super_ldflags = args.pop()
+
+        libs = LibraryList([])
+        flags = defaultdict(list)
+
+        # Check for DSL variants and set corresponding options
+        dsl = self.spec.variants['dsl'].value
+        if dsl != ('none', ):
+            if 'substitute' in dsl:
+                args.append('--enable-py2f=substitute')
+            elif 'verify' in dsl:
+                args.append('--enable-py2f=verify')
+            elif 'serialize' in dsl:
+                raise error.UnsupportedPlatformError(
+                    'serialize mode is not supported yet by icon-liskov')
```

Comment:
I know close to nothing about this, but I thought "liskov" was removed. Is this still relevant?

Reply:
Yes, it is obsolete, so I removed it.